staging/src/k8s.io/component-base/metrics/histogram.go

```diff
@@ -21,17 +21,45 @@ import (
 
 	"github.com/blang/semver/v4"
 	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/trace"
 )
 
 // Histogram is our internal representation for our wrapping struct around prometheus
 // histograms. Summary implements both kubeCollector and ObserverMetric
 type Histogram struct {
+	ctx context.Context
 	ObserverMetric
 	*HistogramOpts
 	lazyMetric
 	selfCollector
 }
 
+type exemplarHistogram struct {
+	*Histogram
+}
+
+func (h *Histogram) Observe(v float64) {
+	h.withExemplar(v)
+}
+
+func (h *Histogram) withExemplar(v float64) {
+	(&exemplarHistogram{h}).withExemplar(v)
+}
+
+func (h *exemplarHistogram) withExemplar(v float64) {
```

Comment:
Why do you need the extra `exemplarHistogram` type and the withExemplar functions.

Reply:
Good catch, I don't. Can't remember why I did that, removed it now.